### PR TITLE
Added common PostSummary component

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
N/A

#### What's this PR do?
Adds a common post summary component. 

*Note:* this component assumes `created` has been parsed to a moment instance.

#### How should this be manually tested?
Passing tests should be sufficient

### CODE REVIEW

File: static/js/factories/posts.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ export const makePost = (isURLPost: boolean = false): Post => ({
   author: `justareddituser${String(Math.random())}`,
   text: isURLPost ? null : casual.text,
   url: isURLPost ? casual.url : null,
+  created: casual.moment,
```
Comment (aliceriot) [inline static/js/factories/posts.js]: will `moment` objects be ok in the redux store? sometimes things with functions defined on them can get a little wacky (I had trouble in the past with Immutable objects, for instance).

File: static/js/components/PostSummary.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// @flow
+import React from 'react';
+
+import type { Post } from '../flow/discussionTypes';
+
+export default class PostItem extends React.Component {
```
Comment (aliceriot) [inline static/js/components/PostSummary.js]: should the class be named `PostSummary` here?

File: static/scss/post.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (aliceriot) [inline static/scss/post.scss]: blank line

File: static/js/factories/posts.js (hunk 1 above)
Comment (rhysyngsun) [inline static/js/factories/posts.js]: Hm, good point.

**Comment by rhysyngsun** [comment]
@aliceriot I've addressed your feedback

File: static/js/factories/posts.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ export const makePost = (isURLPost: boolean = false): Post => ({
   author: `justareddituser${String(Math.random())}`,
   text: isURLPost ? null : casual.text,
   url: isURLPost ? casual.url : null,
-  created: casual.moment,
+  created: casual.date('YYYY-MM-DDThh:mm:ss'),
```
Comment (aliceriot) [inline static/js/factories/posts.js]: I think `casual.moment.format()` would have the same result here, although either way seems fine to me.

**Pull request merged**